core/src/test/java/org/infinispan/util/DeadlockDetectingLockManagerTest.java

```diff
@@ -31,77 +30,72 @@
+         fail("TimeoutException expected!");
+      } catch (TimeoutException e) {
+         //expected
+      }
 
-      assert lockManager.lockAndRecord("k", localTxContext, config.locking().lockAcquisitionTimeout());
-      assert lockManager.getOverlapWithNotDeadlockAwareLockOwners() == 1;
+      AssertJUnit.assertTrue(lockManager.getOverlapWithNotDeadlockAwareLockOwners() > 0); //it's not deterministic. can be 1 or more.
    }
 
    public void testLocalDeadlock() throws Exception {
-      final DldGlobalTransaction ddgt = (DldGlobalTransaction) TransactionFactory.TxFactoryEnum.DLD_NORECOVERY_XA.newGlobalTransaction();
-
-      InvocationContext localTxContext = buildLocalTxIc(ddgt);
+      final DldGlobalTransaction ddgt = new DldGlobalTransaction();
+      final DldGlobalTransaction lockOwner = new DldGlobalTransaction();
 
       ddgt.setCoinToss(0);
       lockOwner.setCoinToss(1);
-      assert ddgt.wouldLose(lockOwner);
-
-      //this makes sure that we cannot acquire lock from the first try
-      Lock mockLock = mock(Lock.class);
-      when(lc.acquireLock(localTxContext.getLockOwner(), "k", SPIN_DURATION, TimeUnit.MILLISECONDS)).thenReturn(null).thenReturn(mockLock);
       lockOwner.setRemote(false);
-      lockOwner.setLockIntention("k");
+      lockOwner.setLockIntention(Collections.singleton("k"));
+      AssertJUnit.assertTrue(ddgt.wouldLose(lockOwner));
+
       lockManager.setOwner(lockOwner);
       lockManager.setOwnsLock(true);
+
+      lockManager.lock("k", lockOwner, 0, TimeUnit.MILLISECONDS).lock();
+
       try {
-         lockManager.lockAndRecord("k", localTxContext, config.locking().lockAcquisitionTimeout());
+         lockManager.lock("k", ddgt, 1000, TimeUnit.MILLISECONDS).lock();
          assert false;
       } catch (DeadlockDetectedException e) {
          //expected
       }
-      assertEquals(1l,lockManager.getDetectedLocalDeadlocks());
+      AssertJUnit.assertEquals(1, lockManager.getDetectedLocalDeadlocks());
+   }
+
+   private Configuration createConfiguration() {
+      Configuration configuration = mock(Configuration.class, RETURNS_DEEP_STUBS);
+      when(configuration.deadlockDetection().spinDuration()).thenReturn(SPIN_DURATION);
+      when(configuration.jmxStatistics().enabled()).thenReturn(true);
+      return configuration;
    }
 
-   private InvocationContext buildLocalTxIc(final DldGlobalTransaction ddgt) {
-      return new LocalTxInvocationContext(mock(LocalTransaction.class)) {
-         @Override
-         public Object getLockOwner() {
-            return ddgt;
-         }
-      };
+   private LockContainer createLockContainer() {
+      PerKeyLockContainer lockContainer = new PerKeyLockContainer(32, AnyEquivalence.getInstance());
+      lockContainer.inject(TIME_SERVICE);
+      return lockContainer;
    }
 
    public static class DeadlockDetectingLockManagerMock extends DeadlockDetectingLockManager {
```

Comment:
Using a mock for the class that's being tested looks odd, I'm wondering if we could remove it.

I tried to remove it myself, but then the real DeadlockDetectingLockManager gave me a `TimeoutException` instead of a `DeadlockDetectedException` in `testLocalDeadlock()`.